src/elements/ConstF.H

```diff
@@ -117,23 +155,15 @@ namespace impactx::elements
             amrex::ParticleReal pyout = py;
             amrex::ParticleReal ptout = pt;
 
-            // length of the current slice
-            amrex::ParticleReal const slice_ds = m_ds / nslice();
-
-            // intermediate quantities - to avoid division by zero
-            amrex::ParticleReal const sincx = (m_kx > 0) ? std::sin(m_kx*slice_ds)/m_kx : slice_ds;
-            amrex::ParticleReal const sincy = (m_ky > 0) ? std::sin(m_ky*slice_ds)/m_ky : slice_ds;
-            amrex::ParticleReal const sinct = (m_kt > 0) ? std::sin(m_kt*slice_ds)/m_kt : slice_ds;
-
             // advance position and momentum
-            xout = std::cos(m_kx*slice_ds)*x + sincx*px;
-            pxout = -m_kx * std::sin(m_kx*slice_ds)*x + std::cos(m_kx*slice_ds)*px;
+            xout = m_cos_kxds * x + m_sincx * px;
+            pxout = m_const_x * x + m_cos_kxds * px;
 
-            yout = std::cos(m_ky*slice_ds)*y + sincy*py;
-            pyout = -m_ky * std::sin(m_ky*slice_ds)*y + std::cos(m_ky*slice_ds)*py;
+            yout = m_cos_kyds * y + m_sincy * py;
+            pyout = m_const_y * y + m_cos_kyds * py;
 
-            tout = std::cos(m_kt*slice_ds)*t + sinct/(betgam2)*pt;
-            ptout = -(m_kt*betgam2) * std::sin(m_kt*slice_ds)*t + std::cos(m_kt*slice_ds)*pt;
+            tout = m_cos_ktds * t + m_const_pt * pt;
+            ptout = m_const_y * t + m_cos_ktds * pt;
```

Comment:
We could actually give those members proper `m_Rij` names...

Reply:
We will need to return the linear map for envelope tracking anyway, in a separate function within `ConstF`.  Those matrix element values could be used here.

Reply:
Thinking about it, I would actually do it the other way around: using the constants in the matrix. Can discuss tomorrow for details.